docs/building-with-stdlib.md

```diff
@@ -0,0 +1,192 @@
+# Building with Standard C++ Library
+
+Standard Library build flavor works best for statically linking the SDK in a
+process (environment where ABI compat is not a requirement), or for
+"header-only" implementation of SDK.
+
+Proposed approach cannot be employed for shared libs in environments where
+ABI compatibility is required:
+
+- OpenTelemetry SDK binary compiled with compiler A + STL B
+- may not be ABI compatible with the main executable compiled with compiler
```

Comment:
I think we have to say "will not" here vs. "may not".

Reply:
I'll change to `will not` .. It's more stable now for vs2015-vs2017-vs2019, as at least those 5 compilers have a promise of ABI compatibility. So on Windows it's more `may not` and on Linux between gcc-4.x and gcc-9x+ it's `will not`.